# Google Drive connection

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


### Complete Linux sandbox

In [2]:
# Current directory
!pwd
# List contents
!ls
# Root
!ls /
# Home
!ls ~/
# Copy: cp, make directory: mkdir, ...

/content
drive  sample_data
bin			    datalab  kaggle  libx32		       opt	   run	 tmp
boot			    dev      lib     media		       proc	   sbin  tools
content			    etc      lib32   mnt		       python-apt  srv	 usr
cuda-keyring_1.0-1_all.deb  home     lib64   NGC-DL-CONTAINER-LICENSE  root	   sys	 var


In [3]:
!ls /content/

drive  sample_data


In [4]:
!ls ~/

### Ready with Python, Tensorflow, Keras, ...

In [5]:
!python --version
# import <tab>

Python 3.10.12


# 1. Create the environment with TensorFlow, Keras, and extras
conda create -n dat300-env python=3.10 tensorflow matplotlib scikit-learn pandas jupyter ipykernel -y

# 2. Activate the environment
conda activate dat300-env

# 3. Register the environment as a Jupyter kernel
python -m ipykernel install --user --name=dat300-env --display-name "Python (dat300-env)"

# 4. Launch Jupyter Notebook
jupyter notebook


## Colab tips

In [18]:
# Check which GPU you got
!nvidia-smi -L

GPU 0: Tesla T4 (UUID: GPU-75f90a7a-8b76-4e38-5649-01e2141675d7)


In [19]:
# Check current resource allocation
!nvidia-smi

Tue Sep  3 19:10:54 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.104.05             Driver Version: 535.104.05   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  Tesla T4                       Off | 00000000:00:04.0 Off |                    0 |
| N/A   44C    P8               9W /  70W |      0MiB / 15360MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+--

In [20]:
# CPU type
!lscpu |grep 'Model name'

Model name:                           Intel(R) Xeon(R) CPU @ 2.20GHz


In [21]:
# CPU information
!lscpu

Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          46 bits physical, 48 bits virtual
  Byte Order:             Little Endian
CPU(s):                   2
  On-line CPU(s) list:    0,1
Vendor ID:                GenuineIntel
  Model name:             Intel(R) Xeon(R) CPU @ 2.20GHz
    CPU family:           6
    Model:                79
    Thread(s) per core:   2
    Core(s) per socket:   1
    Socket(s):            1
    Stepping:             0
    BogoMIPS:             4399.99
    Flags:                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 cl
                          flush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc re
                          p_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3
                           fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand
                           hypervisor lahf_lm abm 3dnowprefetch i

## Time-consuming calculations
- Connect to Google Drive
- Save checkpoints
- Test on something manageble